### PR TITLE
Retain outputs tag if set in config

### PR DESCRIPTION
**Pull request by ComlOnline**
Added the option for messages to be retained.

It works on my install and I think I've made the changes correctly but really not sure.

Cheers

### CODE REVIEW

**Comment by flyte** [comment]
Hey @ComlOnline thanks for the PR! It looks correct to me.

Before I merge it though, would you mind making one change? As far as I'm aware you should just be able to use `output_config["retain"]` by itself in the call to `client.publish()` as it's expecting a boolean value and the Cerberus config validation should ensure that it's the right type already. I'll put a comment on the line I'm talking about.

File: pi_mqtt_gpio/server.py [hunk 1]
```diff
@@ -124,8 +124,10 @@ def set_pin(output_config, value):
         output_config["name"],
         set_value)
     payload = output_config["on_payload" if value else "off_payload"]
+    retain = 1 if output_config["retain"] else 0
```
Comment (flyte) [inline pi_mqtt_gpio/server.py]: Remove this line completely.

File: pi_mqtt_gpio/server.py [hunk 2]
```diff
@@ -124,8 +124,10 @@ def set_pin(output_config, value):
...
     client.publish(
         "%s/%s/%s" % (topic_prefix, OUTPUT_TOPIC, output_config["name"]),
+        retain=retain,
```
Comment (flyte) [inline pi_mqtt_gpio/server.py]: Change this to:

```python
        retain=output_config["retain"],
```

File: README.md [hunk 3]
```diff
@@ -40,6 +40,7 @@ digital_outputs:
     on_payload: "ON"
     off_payload: "OFF"
     initial: low  # This optional value controls the initial state of the pin before receipt of any messages from MQTT. Valid options are 'low' and 'high'.
+    retain: yes # This option value contols if the message is retained. Default is no.
```
Comment (flyte) [inline README.md]: Typo in 'controls'

**Review by flyte — CHANGES_REQUESTED** [review]

**Comment by ComlOnline** [comment]
Hi @flyte thanks for the feedback,

The changes have been made. I didn't know you could use yes/no as boolean values in python, you lean some thing new every day.

Cheers

**Comment by flyte** [comment]
So what's happening here is that the YAML parser picks up the `yes` and `no`, among quite a few other values (as long as they're not within quotes) as boolean values. You can see a list of the values in the "Regexp" section of [this page](http://yaml.org/type/bool.html). This, along with the fact that we're using Cerberus to validate the config, means that we can be sure that the value we get from the dict is a boolean.

**Comment by ComlOnline** [comment]
Ah thank you for explaining that, it will definitely help down the road.

**Pull request merged**